packages/frontend/src/components/dialogs/EditAccountAndPasswordDialog.tsx

```diff
@@ -102,15 +103,29 @@ function EditAccountInner(onClose: DialogProps['onClose']) {
         ]),
       })
 
-      if (confirmed) {
-        update()
+      if (!confirmed) {
+        return
+      }
+    } else if (
+      initial_settings.proxy_enabled !== accountSettings.proxy_enabled &&
+      accountSettings.proxy_enabled === '1'
+    ) {
+      const qr = await BackendRemote.rpc.checkQr(
```

Comment:
are all proxies we support also supported to be read as qr code? Like is the format on both fields the same type?

Also this looks like a workaround, what do the other platforms do? Is there a reason core does not have a `validate_proxy` or even `test_proxy` method?

Reply:
> are all proxies we support also supported to be read as qr code? 
well it's in fact just a string containg (possibly) an URL.

we could do some URL validation ourselves, but I assumed it's better to use the same logic as core.

Reply:
> Also this looks like a workaround, what do the other platforms do?

I discussed it with @adbenitez - he proposed the QR code check.

Reply:
@link2xt is the QR code check sufficient? Is there a better core check available?